### PR TITLE
Use YAML stages

### PR DESCRIPTION
**Pull request by wtgodbe**
Resolves aspnet/AspNetCore#11924 and https://github.com/dotnet/arcade/blob/master/Documentation/CorePackages/YamlStagesPublishing.md for EF6. @dougbu @JunTaoLuo should I port this to any of the release branches (once I have it working)?

@JohnTortugo @bricelam PTAL

Here's the internal build, which looks to be using stages just fine: https://dev.azure.com/dnceng/internal/_build/results?buildId=300829

### CODE REVIEW

**Comment by wtgodbe** [comment]
New internal build w/ these changes rebased against rel/6.3: https://dnceng.visualstudio.com/internal/_build/results?buildId=301201

**Comment by wtgodbe** [comment]
New build with the dup'd commits + channel mapping: https://dnceng.visualstudio.com/internal/_build/results?buildId=301574

**Comment by wtgodbe** [comment]
Internal build looks good. @riarenas @johntortugo any last concerns, or is this good to go?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -9,91 +9,103 @@ variables:
     value: true
   - name: _DotNetArtifactsCategory
     value: ASPNETENTITYFRAMEWORK6
+  - name: _DotNetValidationArtifactsCategory
```
Comment (riarenas) [inline azure-pipelines.yml]: This can be removed if you're satisfied with your testing

**Pull request merged**